conf/defaults.ini

```diff
@@ -1105,6 +1105,26 @@ min_interval = 10s
 # (concurrent queries per rule disabled).
 max_state_save_concurrency = 1
 
+[unified_alerting.external_alertmanager]
+
+# Disable the internal Alertmanager and use an external one instead.
+# The default value is `false`.
+disable_internal_alertmanager = false
+
+# URL of the external Alertmanager that will replace the internal one.
+# Required if `disable_internal_alertmanager` is set to `true`.
+main_alertmanager_url =
+
+# Optional tenant ID to attach to requests sent to the Alertmanager.
+main_alertmanager_tenant_id = 
+
+# Optional username for basic authentication on requests sent to the external Alertmanager.
+# Can be left blank to disable basic auth.
+main_alertmanager_basic_auth_username =
+
+# Optional password for basic authentication on requests sent to the external Alertmanager.
+main_alertmanager_basic_auth_password =
```

Comment:
I saw two downsides to that approach:
- We could have multiple Alertmanager data sources, so we can't know which one of those is to be used as the "main" one.
- We're assuming that the desired Alertmanager is always provisioned as a data source, which might not be the case for OSS users.

Since all fields are optional except for `main_alertmanager_url`, I thought we could use that to identify which of the provisioned Alertmanager data sources the user wants to use as a replacement for the internal one. So, if we only got a URL, we look in the configured data sources for an Alertmanager with that URL and use its credentials.

Reply:
@alexweav I created [this PR](https://github.com/grafana/grafana/pull/73310) using the approach you suggest.